qiime2/sdk/parallel_config.py

```diff
@@ -69,38 +52,49 @@ def setup_parallel(config_fp=None):
         config_fp = _get_vendored_config()
 
     if config_fp is not None:
-        config_dict = get_config(config_fp)
-        mapping = get_mapping(config_dict)
+        parallel_config, mapping = get_config(config_fp)
 
         # If config_dict is empty now, they gave a file that only contained a
         # mapping, so we want to load a default config assuming they do not
         # already have a loaded config
-        if config_dict == {} and PARALLEL_CONFIG.parallel_config is None:
+        if parallel_config is None and PARALLEL_CONFIG.parallel_config is None:
             config_fp = _get_vendored_config()
-            config_dict = get_config(config_fp)
-
-        # Now if we actually have a config dict, we want to load the config. We
-        # still will not have one if they gave us a file that only contained a
-        # mapping while already having a config set up.
-        if config_dict != {}:
-            processed_config = _process_config(config_dict)
-            config = Config(**processed_config)
+            parallel_config, = get_config(config_fp)
 
     # We only want to clear the config if the config we are trying to load is
     # actually different. If we clear the config then load the same config
     # while in the middle of doing something, we're going to have problems. If
     # someone is trying to change the config in the middle of doing something,
     # they are doing things wrong (probably forgot to resolve their future
     # inside of their context manager).
-    if PARALLEL_CONFIG.parallel_config != config:
+    if PARALLEL_CONFIG.parallel_config != parallel_config:
+        # If a config was already loaded, clean up the DFK and clear it for a
+        # new one. If there wasn't already a config loaded, we get an error
+        # which we except and ignore
+        try:
+            dfk = parsl.dfk()
+            dfk.cleanup()
```

Comment:
This has mitigated the socket leak issue, but I am still getting a "too many open files" error on OSX on the Mac Mini